service/resource/legacyv2/adapter/instance.go

```diff
@@ -0,0 +1,116 @@
+package adapter
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/aws-operator/service/keyv2"
+)
+
+// template related to this adapter: service/templates/cloudformation/instance.yaml
+
+type instanceAdapter struct {
+	MasterAZ                     string
+	MasterIAMInstanceProfileName string
```

Comment:
The other fields need the Master prefix to avoid collisions. If this looks OK I'll do a PR to prefix the Worker fields for the launch config & ASG.

Reply:
Totally, better than qualifying the fields IMO.

Reply:
Good point I'm going to keep the Master prefix for the AZ. I'll do a PR to prefix the worker fields.